src/corona_hakab_model/supervisor.py

```diff
@@ -269,3 +275,21 @@ def get(self, manager) -> float:
 
     def name(self) -> str:
         return "Total(" + ", ".join(n.name() for n in self.inners)
+
+class _EffectiveR0Supervisable (FloatSupervisable):
+    def __init__(self):
+        super().__init__()
+
+    def get(self, manager) -> float:
+        # note that this calculation is VARY heavy (that's why it is only calculating once every 5 days)
+        suseptable_indexes = [index for index, val in enumerate(manager.susceptible_vector) if val]
+        return np.sum(1 - np.exp(item) for item in manager.matrix.matrix[suseptable_indexes, :].data) * manager.matrix.total_contagious_probability / manager.matrix.size
```

Comment:
calling np.sum on a generator destroys all of np's power
`np.sum(1 - np.exp(manager.matrix.matrix[suseptable_indexes].data))`?

Reply:
done. improved by a lot